### PR TITLE
[Feature] Support custom worker threads number for iceberg

### PR DESCRIPTION
**Pull request by caneGuy**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

we can set iceberg.worker.num-threads for  ThreadPools.getWorkerPool() below
```
public CloseableIterable<FileScanTask> planFiles(TableOperations ops, Snapshot snapshot, Expression rowFilter, boolean ignoreResiduals, boolean caseSensitive, boolean colStats) {
        ManifestGroup manifestGroup = (new ManifestGroup(ops.io(), snapshot.dataManifests(), snapshot.deleteManifests())).caseSensitive(caseSensitive).select(colStats ? SCAN_WITH_STATS_COLUMNS : SCAN_COLUMNS).filterData(rowFilter).specsById(ops.current().specsById()).ignoreDeleted();
        if (ignoreResiduals) {
            manifestGroup = manifestGroup.ignoreResiduals();
        }

        if (PLAN_SCANS_WITH_WORKER_POOL && snapshot.dataManifests().size() > 1) {
            manifestGroup = manifestGroup.planWith(ThreadPools.getWorkerPool());
        }

        return manifestGroup.planFiles();
    }
```

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 4 / 6 (66.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|2|4|50.00%|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java [hunk 1]
```diff
@@ -21,8 +22,17 @@
...
 import java.util.Map;
 import java.util.Optional;
+import java.util.Properties;
 
 public class IcebergUtil {
+    
+    static {
+        if (Config.iceberg_enable_custom_worker_thread) {
+            Properties props = System.getProperties();
+            props.setProperty("iceberg.worker.num-threads", String.valueOf(Config.iceberg_worker_num_threads));
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java:32]: How about using 'SystemProperties.WORKER_THREAD_POOL_SIZE_PROP'?

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java [hunk 2]
```diff
@@ -21,8 +22,17 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.Properties;
 
 public class IcebergUtil {
+    
+    static {
+        if (Config.iceberg_enable_custom_worker_thread) {
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java:30]: I think this configure can't work as you expected.
When this block is executed, this value would not be initialized, and this value isn't determined by config file.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 4 / 6 (66.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|2|4|50.00%|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 5 / 8 (62.50%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|3|6|50.00%|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 7 / 10 (70.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergRepository.java|4|7|57.14%|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|1|1|100.00%|

**Review by imay — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Pull request merged**